Bull rework part 2

Clay500z opened the pull request:
## About The Pull Request

Tl;dr Bull is no longer stun and knockback immune. All charges now deal 48 damage (2 slashes) exactly. Headbutt has even less stun. Bull's walking speed very slightly increased.

Stun immunity: **Gone**. It can be stunned by buckshot/slugs again, and knocked around by explosives.
Gore: 62 > **48 damage**. It now also does 1 second of slow alongside the stagger. Also also removed it's extra slash that (I think) made it more likely to frac and IB due to wacky code.
Headbutt: 20 > **48 damage**. 2 > **1.5 second stun**.
Plow: 20 > **48 damage**.

## Why It's Good For The Game

**Stun immunity:** This was given to bull as a sort of bandaid for being a terrible caste that couldn't accomplish anything worthwhile with its abilities. Now that its abilities are actually good, its no longer needed. Stun immunity makes Bull too safe in the hands of a good player.

**Gore:** It had a little too much damage, so it's been reduced. The 1 second slow addition doesn't really do much, I'm just adding it for consistencies sake.

**Headbutt:** The stun felt mostly balanced before. But since damage is being increased, stun is being decreased.

**Plow:** This did too little damage to be worth using before, so damage increased.

**Speed:** Slightly increased walking speed is there to compensate for no stun immunity. Should make getting stunned out of a charge less cbt.

Also, gore charging someone now longer makes the "*Gore" sound effect. It mislead marines into thinking they got organ damage, when really it was just a wacky sound effect. This should also make the injection sound effect easier to hear, letting the marine know they've been injected with Neuro. I haven't deleted the sound files themselves as they do sound cool and could be used for something else.

## Changelog

:cl:
balance: All of Bull's charges now deal exactly the same damage (2 slashes).
balance: Gore now also deals 1 sec of slow in addition to the 1 sec of stagger.
balance: Decreased Headbutt charges stun from 2 seconds to 1.5 seconds.
balance: Slightly increased Bulls walking speed.
sounddel: Removed the *gore emote from Bulls gore charge
spellcheck: Updated Bulls ability descriptions
/:cl:

Lefunny commented:
I genuinely think that is more of an overall nerf to the caste? 15 neuro is not worth the massive damage nerf gore took, headbutt should not have the same damage of gore, not only its illogical ((you are piercing a marine with your horns in gore)) but also takes away the characteristic of the attack, gore being the damage rush, headbutt being stun and plow idk what the fuck it was meant to be.

Clay500z commented:
>15 neuro is not worth the massive damage nerf gore took
I think it'll be fine. The damage nerf is actually closer to a 33% reduction instead of a 50% reduction like it seems. Because you can immediately slash again on harm intent after landing a gore, you technically got 4 slashes of damage before. Now it's 3 slashes in total. Neuro plus 1 second of stagger should serve as a good opening charge.

>headbutt should not have the same damage of gore, not only its illogical
I could argue that a blunt hit to the head is equally as deadly as getting impaled, but it's not like we play tgmc for it's logic.

Gore is for harrasing and opening into slashes, headbutt is for stunning and fishing for teamwork, Plow is for crowd control against a line of Ungas.

Ragecrabbo commented:
Going from 42 damage difference between gore and plow to having the same damage seems like a pretty big balance change. I also dont see 15u neuro (roughly same as being spat on by prae once) being worth it to use over a stun. 
I personally think the changes are alright but the numbers might need adjusting a bit.

Clay500z commented:
>I also dont see 15u neuro (roughly same as being spat on by prae once)

Oh whar 💀 I thought prae did like 5u neuro per spit, so 15u seemed good. Guess I'll properly test the neuro amounts other castes add and adjust it accordingly

Araulius commented:
vali mope pr

Clay500z commented:
> vali mope pr


https://github.com/tgstation/TerraGov-Marine-Corps/assets/128223413/3012dc0c-adb6-45c0-bfa0-71b48136a57a

Clay500z commented:
Aight, so I was way off on how much neuro I thought is good. I guess I never really bothered to scan myself after getting neuroed and just took dylo. 
Prae spit adds 15u Neuro + gas cloud 
Boiler lance adds 60u Neuro + gas cloud
Carrier Neuro huggers add 10u Neuro per slash
Defiler adds 15u Neuro per reagent slash

So I've increased Bulls Neuro injection to 30u.

ivanmixo commented on code/modules/mob/living/carbon/xenomorph/charge_crush.dm:
```diff
@@ -603,8 +604,6 @@
 			if(world.time > charge_datum.next_special_attack)
 				charge_datum.next_special_attack = world.time + 2 SECONDS
-				attack_alien_harm(charger, charger.xeno_caste.melee_damage * charger.xeno_melee_damage_modifier, charger.zone_selected, FALSE, TRUE, TRUE) //Free gore attack.
-				emote_gored()
```
Why remove the emote?

Clay500z replied:
It's mostly about the misinformation, now gore won't do enough damage to justify having such a gnarly sounding emote on it. Chem injection being clearer is a nice side effect. ¯\_(ツ)_/¯ If y'all maints agree on keeping it then I'll add it back

ivanmixo commented on code/modules/mob/living/carbon/xenomorph/charge_crush.dm:
```diff
@@ -564,12 +564,13 @@
 			Paralyze(CHARGE_SPEED(charge_datum) * 2 SECONDS)
 		if(CHARGE_BULL_HEADBUTT)
-			Paralyze(CHARGE_SPEED(charge_datum) * 2 SECONDS)
+			Paralyze(CHARGE_SPEED(charge_datum) * 1.5 SECONDS)
```
This just makes the charge not worth it? What are you gonna do with such a short stun?

Clay500z replied:
Goal of headbutt is to fish marines for other xenos to reel in even further with their own abilities. 

I think this is the most balanced stun time I can give it. 2 seconds makes it too easy to infinitely chain stun back to back, giving no chance for the marine to fight back if they're in a 1 tile hallway (assuming you time the charges correctly).

1.5 seconds is just barely enough time to land a second charge from 6 tiles away, according to my testing. And since the second charge isn't full speed, the stun will be reduced and make it impossible to chain another charge.

ivanmixo replied:
If you somehow manage to find a 1 tile hallway long enough for that to be a realistic scenario I'll give you a million dollars.

Clay500z replied:
> If you somehow manage to find a 1 tile hallway long enough for that to be a realistic scenario I'll give you a million dollars.

Well... Shipside maints. And also 3x3 mazes if the marine gets unlucky and ends up flung right in the middle of the 3x3. Or just Pitbull mazes that have 1 tile charging lanes built especially for charging/boiler, but those are rare. 
Also Bull can theoretically keep juggling the marine back forth by swapping harm and help intent. Requires skill and timing, but not so much that its unrealistic.

ivanmixo commented on code/modules/mob/living/carbon/xenomorph/charge_crush.dm:
```diff
@@ -564,12 +564,13 @@
 			adjust_stagger(CHARGE_SPEED(charge_datum) * 1 SECONDS)
-			reagents.add_reagent(/datum/reagent/toxin/xeno_ozelomelyn, 10)
+			adjust_slowdown(CHARGE_SPEED(charge_datum) * 1)
+			reagents.add_reagent(/datum/reagent/toxin/xeno_neurotoxin, 30)
```
On one hand 30u neuro is insane, but 10u ozy is arguably more powerful and balanced. Getting a gore charge off on someone twice is not hard and 60u neuro is extremely debilitating.

Clay500z replied:
It's almost 2 prae spits worth (lesser since prae also leaves a gas cloud). And I think staying in the Bulls charging lane after getting gored once can be chalked up to an issue of skill.

Only place I can see it being too strong is when xenos surround a marine inside maze, making the gore charges guaranteed. But then again, xenos surrounding marine is usually death within seconds anyways.

Then again, I'm not an expert on neuro. I didn't even know Prae did 15u before making this pr. Maybe 25u would be better?

shelteredfreak commented:
As much as I love toxins, I don't think dropping the damage of Gore (Single-target) and raising the damage of plow (multi-target) to be the same is exactly consistent with Bull's design principle. In practice, Gore doesn't do as much damage as you think it does. And the amount of damage it can do is directly proportionate to the bulls speed which is tied to the time spent charging, so it's not always the maximum damage anyhow.

Don't drop the gore damage that low just to make Bull into a discount Prae using neuro. Especially not making it do the same amount of damage as Plow. You sacrifice all of your speed instantly and risk getting shotgunned out of your charge lane because you immediately stop in your tracks once you Gore a marine. If anything I think Gore should get a bit more damage to be honest, but that's just me.

I just like the utility of Bull being able to use all its charges differently and have unique charges for each situation. I don't want to see them start melding together until it just gets made into one charge and everyone calls it discount Crusher which it already is seen as. 

Gore is for high single-target burst damage, Headbutt is for marine displacement (and combo'ing with other xenos), and Plow is just Crusher version 0.5. Don't lose this uniqueness.

Clay500z commented:
>In practice, Gore doesn't do as much damage as you think it does. And the amount of damage it can do is directly proportionate to the bulls speed

This shouldn't be the case anymore, it was Gore's wacky snowflake slash that was affected by speed, and I've removed and replaced that for normal damage that isn't affected by speed. Only Headbutts stun and Gore's stagger is affected by speed now.

>I just like the utility of Bull being able to use all its charges differently and have unique charges for each situation

That's precisely why I equalized the damage on all charges, so that you aren't just locked to using the charge that deals the highest damage and ignore the other two. Gore siill does 1 extra slash of damage + Neuro, so it should theoretically still be the best single target damage.

> Discount Prae 
> Discount Crusher

Who cares, man. What matters is that they're fun and effective to play. We aren't deleting Spitter for being discount Prae, Pupeteer for being discount Widow, or runner for being discount Hunter either. They take up different tier slots and are different enough to play.

chizzy376 commented:
> I genuinely think that is more of an overall nerf to the caste? 

The "first part of the rework" was just major powercreep imo

Clay500z commented:
> The "first part of the rework" was just major powercreep imo

Yeah, it was. I don't think anyone was really sure exactly how strong/weak Bull even was, simply because literally no one was playing it after the primo nerf. So it was kind of a shot in the dark on how I should be buffing it. Now that people are actually playing Bull again, I have a better idea of what to tweak and by how much.

Kuro020 reviewed (changes requested):
I'm not gonna lie, this feels like a flat out nerf to the caste, why is gore doing the same damage as plow when plow hits everyone and lets you keep charging?

Trading ozy for neuro is a big oof as well

Kuro020 reviewed (commented):
just make it ozy again (or some other xeno chem) and we can test it, I dont think neuro really fits here

ivanmixo commented on code/modules/mob/living/carbon/xenomorph/castes/bull/abilities_bull.dm:
```diff
@@ -28,7 +28,7 @@
 	name = "Gore Charge"
 	action_icon_state = "bull_gore"
-	desc = "When you hit a host, stops your charge while piercing and injecting them with Neurotoxin."
+	desc = "When you hit a host, stops your charge while piercing and injecting them with Ozelomelon."
```
melon

Clay500z replied:
MY HEART SAYS ITS MELON!!!

Kuro020 reviewed (approved):
should be ok for testing

chemistrymain3 commented:
I struggle to understand why bull needs a nerf in any form.

The no stun immunity is insane, by the way. You just stand in its lane with a shotgun and it's basically fucked.
You've slightly buffed the charge that passes through people and nerfed everything else. I can't see a reason why anyone would pick this caste, now that it's basically lost all suppressive ability or utility that it previously had when it was already pretty weak-

There's nothing the bull does, *especially* after these changes, that another caste doesn't do better.

Clay500z commented:
> I struggle to understand why bull needs a nerf in any form.

Deleting the stun immunity was needed imo, it made the caste a little too safe. Plow was buffed and Headbutt got a sidegrade. Reducing Gores damage might have been too much though. It's TMed, so we'll see.
